File: KeeperFINAL.client/src/models/Vault.js
import { Account } from "./Account.js"
export class Vault {
  constructor(data) {
    this.id = data.id
    this.creatorId = data.creatorId
    this.creator = new Account(data.creator)
    // this.profile = data.profile
    this.createdAt = data.createdAt
    this.updatedAt = data.updatedAt
    this.name = data.name
    this.description = data.description
    this.img = data.img
    this.keepCount = data.keepCount
    this.viewCount = data.views
    this.isPrivate = data.isPrivate
  }
}

File: KeeperFINAL/Models/Vault.cs
namespace KeeperFINAL.Models;
using KeeperFINAL.Interfaces;
using System.ComponentModel.DataAnnotations;

public class Vault : IHasCreator
{
  public int Id { get; set; }
  [Required]
  public string Name { get; set; }
  public string Description { get; set; }
  public string Img { get; set; }
  public int KeepCount { get; set; }
  public bool? IsPrivate { get; set; } = false;

  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
}